### PR TITLE
Fastqe: escape element identifier

### PR DESCRIPTION
**Pull request by mblue9**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by hexylena** [comment]
Can you add a `+galaxy0` to the version since the behaviour is changing here?

**Comment by mblue9** [comment]
Thanks @hexylena! I didn't update the wrapper version as I had thought of it as a bug fix but yes it does alter the output. Thanks for making the change @bgruening!
Do you know why the test is failing here or how I can see the what the error is?
In the log I just see `Unsuccessful tests found, inspect the 'All tool test results' artifact for details.`

**Comment by bgruening** [comment]
You can see the error here https://github.com/galaxyproject/tools-iuc/pull/3141/checks?check_run_id=903659685 in the "planemo tool test". Seems that the test-data changed?

**Comment by mblue9** [comment]
ah ok I think I understand, the output now contains text like _fq instead of .fq but I couldn't update the test output well as sed OSX works differently, @hexylena or someone are you able to update the test data using Linux or do you know what I need to do to make that sed command also OSX compatible & I can update the test data?

**Comment by bgruening** [comment]
One thing to make it compatible with OSX is to include sed as a requirement for the tool.

**Comment by hexylena** [comment]
image

we could permit `.` to fix the failing test.

If the sed command doesn't work on OSX, I'm sorry. I don't have OSX and can't test what sed commands are platform independent :( But bjoern is right, adding sed should help.

File: tools/fastqe/fastqe.xml [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+        <requirement type="package" version="4.7">sed</requirement>
     </requirements>
     <command detect_errors="exit_code">
     <![CDATA[
 
+    #import re
+
     #for i in $input:
-        ln -s '$i' '$i.element_identifier' &&
+        #set $filename = re.sub('[^\w\-\s]', '_', str($i.element_identifier))
```
Comment (hexylena) [inline tools/fastqe/fastqe.xml:13]:
```suggestion
        #set $filename = re.sub('[^\w\-\s.]', '_', str($i.element_identifier))
```

File: tools/fastqe/fastqe.xml [hunk 2]
```diff
@@ -19,7 +23,8 @@
         --long $long
         --max
         #for i in $input:
-            '$i.element_identifier'
+            #set $filename = re.sub('[^\w\-\s]', '_', str($i.element_identifier))
```
Comment (hexylena) [inline tools/fastqe/fastqe.xml:26]:
```suggestion
            #set $filename = re.sub('[^\w\-\s.]', '_', str($i.element_identifier))
```

**Comment by mblue9** [comment]
Thanks for the sed tip @bgruening and looking into it @hexylena ! Yep adding sed as requirement seems to work great 👍

**Comment by hexylena** [comment]
ah you've updated the test data that works too, no need to accept my changes then.

File: tools/fastqe/fastqe.xml (hunk 2 above)
Comment (mblue9) [reply tools/fastqe/fastqe.xml:26]: Oh just saw this, should I allow the "." or let it change to underscore?

Comment (hexylena) [reply tools/fastqe/fastqe.xml:26]: Doesn't really matter to me. would `.`.be nice in the identifiers forusers?

Comment (mblue9) [reply tools/fastqe/fastqe.xml:26]: Maybe a bit nicer to see `.fastq.gz` than `_fastq_gz` but not sure if allowing "." is considered unsafe (ping @nsoranzo)

Comment (hexylena) [reply tools/fastqe/fastqe.xml:26]: it's safe, they're single quoted everywhere so no risk of directory traversal.

Comment (mblue9) [reply tools/fastqe/fastqe.xml]: Ok great, in that case I've allowed the `.`.

Comment (nsoranzo) [reply tools/fastqe/fastqe.xml]: Dots seem fine to me.

File: tools/fastqe/fastqe.xml (hunk 1 above)
Comment (hexylena) [reply tools/fastqe/fastqe.xml:13]: this should match the other one, right? I honestly don't know how this passed tests, if it did `ln -s '...' 'test.fq'` here and then later `fastqe ... 'test_fq'`

Comment (nsoranzo) [reply tools/fastqe/fastqe.xml:13]: In a bit I am going to rewrite this to save the filenames to a list and not repeat ourselvers.

Comment (mblue9) [reply tools/fastqe/fastqe.xml:13]: Thanks @nsoranzo ! I meant to accept both @hexylena and I don't know how that passed tests either.

Comment (mblue9) [reply tools/fastqe/fastqe.xml:13]: Ah actually think I do know how it passed `.` needs to be escaped here?

**Comment by nsoranzo** [comment]
The tests were working when they should have failed and viceversa, but a rebase fixed it (weird!)

File: tools/fastqe/fastqe.xml (hunk 1 above)
Comment (nsoranzo) [reply tools/fastqe/fastqe.xml]: > Ah actually think I do know how it passed `.` needs to be escaped here?

No, according to https://docs.python.org/3/library/re.html#regular-expression-syntax , special characters lose their special meaning inside sets of characters.

**Comment by mblue9** [comment]
Ok thanks for the info & merge @nsoranzo !

**Pull request merged**